src/OneExplorer/OneExplorer.ts

```diff
@@ -174,29 +174,30 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
     let warningMessage;
     if (oneNode.node.type === NodeType.baseModel) {
       // TODO automatically change the corresponding files
-      warningMessage = 'WARNING: You may need to change input paths in the following cfg files:\n';
-      for (const conf of oneNode.node.childNodes) {
-        // NOTE A newline character is not supported in input box.
-        // vscode-extension doesn't plan to support multiple lines inside in input box or in
-        // notification box.
-        warningMessage += `${conf.name} `;
-      }
-    } else {
-      warningMessage = 'WARNING: Renaming may result in some unexpected changes on the tree view.';
+      warningMessage = `WARNING! ${
+          oneNode.node.childNodes.map(node => `'${node.name}'`)
+              .toString()} will disappear from the view.`;
+    } else if (oneNode.node.type === NodeType.derivedModel) {
```

Comment:
Point1. (cfg) Any name is fine for cfg. Changing cfg file won't make a tree change inside the view.
Point2. (basemodel) Basemodel name change will definitely remove config node from the view.
Point3. (derivedmodel) Derived model name change may remove the node itself from the view. (It may disappear or not)